File: snapper.py
import csv

input_file_name = "/home/joranvar/A-large-practice.in"
output_file_name = "output_large.txt"

def solve(n, k):
    if (k+1) % (2**n) == 0:
        return "ON"
    else:
        return "OFF"

with open(input_file_name, 'r') as input_file:
    reader = csv.reader(input_file, delimiter=' ')
    _ = next(reader)
    r = 0
    with open(output_file_name, 'w') as output_file:
        for row in reader:
            r = r + 1
            result = solve(int(row[0]), int(row[1]))
            output_file.write("Case #%s: %s\n" % (r, result))

File: snapper.hs
#! /usr/bin/env nix-shell
#! nix-shell -i runghc -p ghc

inputFileName = "/home/joranvar/A-large-practice.in"
outputFileName = "output_large.txt"

solve :: (Int, Int) -> String
solve (n, k)
    | (k+1) `mod` (2 ^ n) == 0 = "ON"
    | otherwise = "OFF"

main :: IO ()
main = do
  _:rows <- lines <$> readFile inputFileName
  writeFile outputFileName $ concat $ zipWith format [1..] $ map (solve . parse) rows
  where
    parse row = case map read $ words row of
      [n, k] -> (n, k)
      _ -> error $ "Parsing does not give two ints: " ++ show row
    format r result = "Case #" ++ show r ++ ": " ++ result ++ "\n"
